vscode-extension/src/aiConfigEditor.ts

```diff
@@ -0,0 +1,413 @@
+    });
+
+    startServer.on("close", (code) => {
+      if (code !== 0) {
+        this.extensionOutputChannel.error(
+          this.prependMessage(
+            `Server at port=${openPort}, pid=${startServer.pid} was terminated unexpectedly with exit code ${code}`,
+            document
+          )
+        );
+        console.error(`server terminated unexpectedly: exit code=${code}`);
+      } else {
+        this.extensionOutputChannel.info(
+          this.prependMessage(
+            `Server at port=${openPort}, pid=${startServer.pid} was terminated successfully.`,
+            document
+          )
+        );
+        console.log(`server terminated successfully`);
+      }
+    });
+
+    startServer.on("error", (err) => {
+      this.extensionOutputChannel.error(
+        this.prependMessage(JSON.stringify(err), document)
+      );
+      // TODO: saqadri - add "restart" option with error message
+    });
+
+    const serverUrl = `http://localhost:${openPort}`;
+
+    return {
+      proc: startServer,
+      url: serverUrl,
+    };
+  }
+
+  /**
+   * Get the static html used for the editor webviews.
+   */
+  private getHtmlForWebview(
+    webview: vscode.Webview,
+    editorServer: ServerInfo
+  ): string {
+    // The JS file from the React build output
+    const scriptUri = getUri(
+      webview,
+      this.context.extensionUri,
+      "editor",
+      "build",
+      "static",
+      "js",
+      "main.js"
+    );
+
+    // Use a nonce to whitelist which scripts can be run
+    const nonce = getNonce();
+
+    // Tip: Install the es6-string-html VS Code extension to enable code highlighting below
+    return /*html*/ `
+	 <!DOCTYPE html>
+	 <html lang="en">
+	   <head>
+		 <meta charset="utf-8">
+		 <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">
+		 <meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src data: http: https: 'self'; connect-src vscode-webview: ${editorServer.url} http: https:; style-src 'unsafe-inline' ${webview.cspSource} https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0 https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/editor/editor.main.css; script-src 'nonce-${nonce}' vscode-resource: https: http: https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0 https://cdn.jsdelivr.net; font-src https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/base/browser/ui/codicons/codicon/codicon.ttf; worker-src blob:;">
+		 <title>Hello World</title>
+	   </head>
+	   <body>
+		 <noscript>You need to enable JavaScript to run this app.</noscript>
+		 <div id="root"></div>
+		 <script nonce="${nonce}" src="${scriptUri}"></script>
+	   </body>
+	 </html>
+   `;
+  }
+
+  /**
+   * Replaces the entire existing document with the provided new text.
+   */
+  private updateTextDocument(document: vscode.TextDocument, newText: string) {
+    const edit = new vscode.WorkspaceEdit();
+
+    // TODO: saqadri - figure out a way to update the document piecemeal

```

Comment:
Would we ever want each prompt to be its own document? I vaguely remember that being the case in other applications, I think due to LSP integrations?